stellarphot/core.py

```diff
@@ -62,3 +64,307 @@ def __init__(self, gain=1.0 * u.electron / u.adu,
+
+        # Extract appropriate information from input `table_descript`
+        colnames = self._table_descript[:,0].tolist()
+        coltypes = self._table_descript[:,1].tolist()
+        colunits = self._table_descript[:,2].tolist()
+        attrnames = self._table_descript[:,3].tolist()
+
+        # Check that required attributes are in the input attributes
+        required_attr = ['ra', 'dec', 'id']
+        required_units = [u.deg, u.deg, None]
+        for i, this_attr in enumerate(required_attr):
+            if this_attr not in attrnames:
+                raise ValueError(f"Required attribute '{this_attr}' is not defined in your table_descript.")
+            if required_units[i] is not None:
+                if colunits[attrnames.index(this_attr)] != required_units[i]:
+                    raise ValueError(f"Required attribute '{this_attr}' must have units {required_units[i]} (table_descript has '{colunits[attrnames.index(this_attr)]}').")
+
+        # Perform processing once initialized
+        if self.data is None:
+            # Create empty astropy table with right format and data
+            self.data = Table(names=colnames, masked=False, dtype=coltypes, units=colunits)
+        else:
+            # Check the format of the data table matches the table_descript by checking
+            # each column listed in table_descript exists and is he write type and unit.
+            # NOTE: This ignores any columns not in the table_descript, it does not remove them.
+            for i,this_col in enumerate(colnames):
+                # Failure is assumed to be due to bad column value
+                try:
+                    data[this_col]
+                except KeyError:
+                    raise ValueError(f"column {this_col} is not present in input data.")
+                # Check type
+                if data[this_col].dtype != coltypes[i]:
+                    raise ValueError(f"data[{this_col}] is of wrong type (declared {coltypes[i]} but reported as {data[this_col].dtype}).")
+                if data[this_col].unit != colunits[i]:
+                    raise ValueError(f"data[{this_col}] is of wrong unit (declared {colunits[i]} but reported as {data[this_col].unit}).")
+
+        # Create attibutes corresponding to table column
+        self.cols_to_attr(colnames, attrnames)
+
+
+    def cols_to_attr(self, colnames, attrnames):
+        # Connects table columns to attributes
+        for i,attr in enumerate(attrnames):
+            if attr is not None:
+                self.__setattr__(attr, self.data[colnames[i]])
+
+
+class PhotometryData(BaseEnhancedTable):
+    """
+    A base class to hold an `astropy.table.Table` table of reduce photometry
+    data, extending it with a mapping of various table columns to attributes.
+
+    USAGE NOTES: If you input a data file, it MUST contain the following columns
```

Comment:
I think this should be lower in the docstring, in the "Notes" section. I also think we need to auto-generate this.

Reply:
I moved it, but still not my favorite approach.  I do think if you can think of attributes we don't need, go ahead and remove them.

Reply:
oops, I meant move this usage note, not the attributes.  The motivation for having the usage note lower is primarily that it is so long.

Reply:
I did move the usage note  to the end of the parameters.  I wanted to keep it next to the discussion of the `data` parameter since it really is a discussion of the restrictions on that `data` table.